libs/language-server/src/lib/validation/validation-context.ts

```diff
@@ -0,0 +1,26 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import { AstNode, DiagnosticInfo, ValidationAcceptor } from 'langium';
+
+export class ValidationContext {
+  private errorOccurred = false;
```

Comment:
We could also count the `errors`, `warnings`, etc. and then have derived attributes/methods for `hasErrorOccured` / `hasWarningOccured`. Not sure if we need such a thing soon, though.

Reply:
Yeah, also thought about that, but for now a boolean flag is sufficient. Can be added if needed in the future :+1: